packages/reactivity/src/ref.ts

```diff
@@ -105,7 +105,7 @@ class RefImpl<T> {
   private _value: T
   private _rawValue: T
 
-  public dep?: Dep = undefined
+  public dep?: Dep
```

Comment:
In `TypeScript`, if a class member property is not shown for assignment, it will be erased after compilation.

The real point of optimization is probably the `hidden class` of `v8`, which makes the structure of an object static, i.e., its structure is determined at initialization and not dynamically added or deleted to avoid destroying the structure and thus losing the opportunity of optimization.

Of course I don't really understand the specific optimization details of this `hidden class`, and it's not clear why `TS` wants to erase these member properties, depending on how the author (@yyx990803 ) is thinking about it.

---

[TS Babel Playground](https://babeljs.io/repl#?browsers=defaults%2C%20not%20ie%2011%2C%20not%20ie_mob%2011&build=&builtIns=false&corejs=3.21&spec=false&loose=false&code_lz=MYGwhgzhAECC0G8BQ1oAcBOB7NsD8AXNAHYCuAtgEYCmGK62aAQoSRTRtALzSnEAm1AGYBLYtX5IAvkiA&debug=false&forceAllTransforms=false&shippedProposals=false&circleciRepo=&evaluate=false&fileSize=false&timeTravel=false&sourceType=module&lineWrap=true&presets=env%2Cstage-0%2Cstage-1%2Cstage-2%2Cstage-3%2Ctypescript&prettier=false&targets=&version=7.19.0&externalPlugins=&assumptions=%7B%7D)

![image](https://user-images.githubusercontent.com/51586637/188563437-8dbe34b1-ff73-4604-8e81-eaeda2de7da1.png)

---

[JS Babel Playground](https://babeljs.io/repl#?browsers=defaults%2C%20not%20ie%2011%2C%20not%20ie_mob%2011&build=&builtIns=false&corejs=3.21&spec=false&loose=false&code_lz=MYGwhgzhAECC0G8BQ1oAcBOB7NsXuzQCFoBeaAVwDsATAUwDMBLKumpAXySA&debug=false&forceAllTransforms=false&shippedProposals=false&circleciRepo=&evaluate=false&fileSize=false&timeTravel=false&sourceType=module&lineWrap=true&presets=env%2Cstage-0%2Cstage-1%2Cstage-2%2Cstage-3&prettier=false&targets=&version=7.19.0&externalPlugins=&assumptions=%7B%7D)

![image](https://user-images.githubusercontent.com/51586637/188563629-50f2c3ca-c97b-4688-a410-12f546058afa.png)

Reply:
Unlike `_value` and `_rawValue`, which are already initialized inside the `constructor`.

Reply:
Of course, I think we do not have to deliberately do optimization, that is also too tired point.